### DISCUSSION TITLE
opentelemetry collector & opentelemetry api version compatibility

### DISCUSSION THREAD

**Discussion by vinodkumarpandiyan** [open-telemetry/opentelemetry-java#6802] [Q&A] [answered]
Hi,
We are using opentelemetry api version 1.0.0 in one of our library. We are planning to upgrade it to 1.41.0. 

Since we upgrade opentelemetry api & sdk to 1.41.0, we want to know what is the otelcollector compatible version.

I could not find any suitable supporting docs. Hence i contact you here.

Could you please assist?

Thanks

**Comment by jack-berg** [comment, marked as answer]
I assume you're referring to the version of [opentelemetry-proto](https://github.com/open-telemetry/opentelemetry-proto) since `opentelemetry-java` isn't coupled to the version of the open telemetry collector in any way except through OTLP.

`opentelemetry-java` uses [opentelemetry-proto v1.3.2](https://github.com/open-telemetry/opentelemetry-java/blob/v1.41.0/dependencyManagement/build.gradle.kts#L71).

Many opentelemetry-collector versions will support opentelemetry-proto 1.3.2 since protobuf messages have [backwards compatibility](https://github.com/open-telemetry/opentelemetry-proto?tab=readme-ov-file#stability-definition).

**Comment by serv-inc** [comment]
@jack-berg : thanks for the explanation. Since it is a bit tedious to dig through the source code to find the version of opentelemetry-proto, and to link that to the correct opentelemetry-collector, and the correct opentelemetry-javaagent , which both have different versions: is there a list detailing the relationships ?

**Reply by serv-inc** [reply]
Is https://opentelemetry.io/docs/languages/java/intro/#dependencies-and-boms relevant for this ?

**Reply by serv-inc** [reply]
See also https://stackoverflow.com/questions/79354998/opentelemetry-java-matching-versions

**Reply by jack-berg** [reply]
The relationship between `opentelemetry-java` and the version of `opentelemetry-proto` it uses comes up enough that it may warrant special treatment. Would you mind opening an issue for this?

**Reply by serv-inc** [reply]
@jack-berg : done, thanks: https://github.com/open-telemetry/opentelemetry-java/issues/7025
